### ISSUE TITLE
Mutation potential bug - schema validation executed after resolving mutation

### ISSUE THREAD

**Issue by m-naw** [Youshido/GraphQL#142]
When I execute mutation like that one from documentation and I add invalid field in expected response schema

```
mutation {
  createPost(author: "Alex", post: {title: "Hey, this is my new post", summary: "my post" }) {
   title
   invalid_field
  }
}
```

Resolve function is executed (so for example Post is added to database) and error is returned after that

```
{
  "data": {
    "createPost": null
  },
  "errors": [
    {
      "message": "Field \"invalid_field\" not found in type \"Post\"",
      "locations": [
        {
          "line": x,
          "column": y
        }
      ]
    }
  ]
}
```

Probably schema should be validated before mutation is being resolved ? Am I right ?

**Comment by dilab** [comment]
**invalid_field** is the return result of resolver. 

How would the schema validate the result before you return it?

**Comment by m-naw** [comment]
This is why I wrote that its potential bug as I know what u mean. 

But from logical perspective mutation is successfully executed so I shouldnt get an error or schema should be validated before mutation is being resolved.

I dont know how it behaves in other graphql implementations ?

**Comment by fritz-gerneth** [comment]
I think the [specifation](https://facebook.github.io/graphql/#sec-Validating-Requests) is quite clear on this:

> As explained in the Validation section, only requests which pass all validation rules should be executed. If validation errors are known, they should be reported in the list of “errors” in the response and the request must fail without execution.

So I agree that the mutation may not be executed due to known validation errors.

**Comment by viniychuk** [comment]
Thanks all for reporting and participating..
@fritz-gerneth you're right. Even tho it'll require some work because of directives (that are in progress) and fragments... added to the list.

**Comment by portey** [comment]
done, merged to v2.0 (wait for release)
